docs/docs/hydro/live-collections/streams.md

```diff
@@ -3,10 +3,119 @@ sidebar_position: 2
+
+Streams have several type parameters:
+- `T`: the type of elements in the stream
+- `L`: the location the stream is on (see [Locations](../locations/index.md))
+- `B`: indicates whether the stream is [bounded or unbounded](./bounded-unbounded.md)
+- `Order`: indicates whether the elements in the stream have a deterministic order or not
+  - This type parameter is _optional_; by default the order is deterministic
+
+## Creating a Stream
+The simplest way to create a stream is to use the [`source_iter`](https://hydro.run/rustdoc/hydro_lang/location/trait.Location#method.source_iter) method on a location, which creates a stream from any Rust type that can be converted into an [`Iterator`](https://doc.rust-lang.org/beta/std/iter/trait.Iterator.html) (via [`IntoIterator`](https://doc.rust-lang.org/std/iter/trait.IntoIterator.html)). For example, we can create a stream of integers on a [process](../locations/processes.md) and transform it:
+
+```rust
+# use hydro_lang::*;
+# use dfir_rs::futures::StreamExt;
+# tokio_test::block_on(test_util::multi_location_test(|flow, p_out| {
+let process = flow.process::<()>();
+let numbers: Stream<_, Process<_>, _> = process
+    .source_iter(q!(vec![1, 2, 3]))
+    .map(q!(|x| x + 1));
+// 2, 3, 4
+# numbers.send_bincode(&p_out)
+# }, |mut stream| async move {
+# for w in 2..=4 {
+#     assert_eq!(stream.next().await, Some(w));
+# }
+# }));
+```
+
+Streams also can be sent over the network to form distributed programs. Under the hood, sending a stream sets up an RPC handler at the target location that will receive the stream elements. For example, we can send a stream of integers from one process to another with [bincode](https://docs.rs/bincode/latest/bincode/) serialization:
+
+```rust
+# use hydro_lang::*;
+# use dfir_rs::futures::StreamExt;
+# tokio_test::block_on(test_util::multi_location_test(|flow, p_out| {
+let p1 = flow.process::<()>();
+let numbers: Stream<_, Process<_>, _> = p1.source_iter(q!(vec![1, 2, 3]));
+let p2 = flow.process::<()>();
+let on_p2: Stream<_, Process<_>, _> = numbers.send_bincode(&p2);
+// 1, 2, 3
+# on_p2.send_bincode(&p_out)
+# }, |mut stream| async move {
+# for w in 1..=3 {
+#     assert_eq!(stream.next().await, Some(w));
+# }
+# }));
+```
+
+## Stream Ordering and Determinism
+When sending a stream over the network, there are certain situations in which the order of messages will not be deterministic for the receiver. For example, when sending streams from a cluster to a process, delays will cause messages from different cluster members to be interleaved in a non-deterministic order.
+
+To track this behavior, stream have an `Order` type parameter that indicates whether the elements in the stream will have a deterministic order ([`TotalOrder`](pathname:///rustdoc/hydro_lang/stream/struct.TotalOrder)) or not ([`NoOrder`](pathname:///rustdoc/hydro_lang/stream/struct.NoOrder)). When the type parameter is omitted, it defaults to `TotalOrder` for brevity.
```

Comment:
I don't think that last sentence is quite accurate due to type inference. In some cases we infer `NoOrder` without a type parameter being provided by the developer. 

More generally, in the absence of assertions from elsewhere shouldn't we make a conservative inference of `NoOrder`?